### PR TITLE
Convert blocks

### PR DESCRIPTION
**Pull request by ahuang11**
## Summary

Convert credentials into blocks

### CODE REVIEW

File: prefect_azure/credentials.py [hunk 1]
```diff
@@ -64,18 +67,20 @@ def inner(*args, **kwargs):
...
     a connection string.
 
     Args:
         connection_string: includes the authorization information required
     """
 
-    connection_string: str
+    _block_type_name = "Blob Storage Azure Credentials"
+    _logo_url = "https://github.com/PrefectHQ/orion/blob/main/docs/img/collections/azure.png?raw=true"  # noqa
+
```
Comment (desertaxle) [inline prefect_azure/credentials.py:82]: "Azure Blob Storage Credentials" reads a little more naturally, I think. Might make sense to update the class name too?
```suggestion
    _block_type_name = "Azure Blob Storage Credentials"
    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/6AiQ6HRIft8TspZH7AfyZg/39fd82bdbb186db85560f688746c8cdd/azure.png?h=250"  # noqa
```

File: prefect_azure/credentials.py [hunk 2]
```diff
@@ -177,23 +184,25 @@ async def example_get_container_client_flow():
...
     Azure authentication is handled via the `azure` module through
     a connection string.
 
     Args:
         connection_string: includes the authorization information required
     """
 
-    connection_string: str
+    _block_type_name = "Cosmos DB Azure Credentials"
+    _logo_url = "https://github.com/PrefectHQ/orion/blob/main/docs/img/collections/azure.png?raw=true"  # noqa
```
Comment (desertaxle) [inline prefect_azure/credentials.py:203]: Starting with Azure makes sure that they'll all be next to each other in the blocks UI. Maybe update the class name too?
```suggestion
    _block_type_name = "Azure Cosmos DB Credentials"
    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/6AiQ6HRIft8TspZH7AfyZg/39fd82bdbb186db85560f688746c8cdd/azure.png?h=250"  # noqa
```

File: prefect_azure/credentials.py [hunk 3]
```diff
@@ -321,22 +332,23 @@ def get_workspace(self) -> "Workspace":
             @flow
             def example_get_workspace_flow():
                 azure_credentials = MlAzureCredentials(
-                    "tenant_id",
-                    "service_principal_id",
+                    tenant_id="tenant_id",
+                    service_principal_id="service_principal_id",
                     "service_principal_password",
```
Comment (desertaxle) [inline prefect_azure/credentials.py:337]: `service_principal_password = "service_principal_password"`

File: prefect_azure/credentials.py [hunk 4]
```diff
@@ -299,9 +307,12 @@ class MlAzureCredentials:
         workspace_name: The existing workspace name.
     """
 
+    _block_type_name = "AzureML Credentials"
+    _logo_url = "https://github.com/PrefectHQ/orion/blob/main/docs/img/collections/azure.png?raw=true"  # noqa
```
Comment (desertaxle) [inline prefect_azure/credentials.py:311]:
```suggestion
    _block_type_name = "AzureML Credentials"
    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/6AiQ6HRIft8TspZH7AfyZg/39fd82bdbb186db85560f688746c8cdd/azure.png?h=250"  # noqa
```

File: prefect_azure/credentials.py [hunk 5]
```diff
@@ -284,10 +293,9 @@ def example_get_container_client_flow():
         return container_client
 
 
-@dataclass
-class MlAzureCredentials:
+class MlAzureCredentials(Block):
```
Comment (desertaxle) [inline prefect_azure/credentials.py:296]: Update to match block type name?
```suggestion
class AzureMlCredentials(Block):
```

**Review by desertaxle — CHANGES_REQUESTED** [review]

**Comment by ahuang11** [comment]
If we change class name, does this mean the next version should be 1.0.0 since it's technically breaking?

**Comment by desertaxle** [comment]
We're below 1.0.0 right now, which normally implies that the API isn't stable yet. We can bump to 0.2.0 and note breaking changes in the change log.

**Review by desertaxle — APPROVED** [review]

**Pull request merged**